specifications/api/swagger.yaml

```diff
@@ -1628,6 +1628,30 @@ components:
         deposits: *deposits
         metadata: *transactionSerializedMetadata
 
+    ApiWalletMigrationBalance: &ApiWalletMigrationBalance
+      type: object
+      required:
+        - ada
+        - assets
+      properties:
+        ada: *amount
+        assets: *walletAssets
+
+    ApiWalletMigration: &ApiWalletMigration
+      type: object
+      required:
+        - selections
+        - cost
+        - balance_included
+        - balance_leftover
+      properties:
+        selections:
+          type: array
+          items: *ApiCoinSelection
+        cost: *amount
+        balance_included: *ApiWalletMigrationBalance
+        balance_leftover: *ApiWalletMigrationBalance
```

Comment:
> I guess we should explain why there might be leftovers, now or later, such that users know if they need to care.

Definitely!

> Assets could be leftover in some cases, I recall.

That's right. The requirements allow us to omit assets if it the algorithm cannot find a viable strategy to include them. An example of when this might happen is if the remaining UTxO contains a single entry with the minimum ada amount and a single non-ada asset. In this situation, it's impossible to create a transaction, because there's not enough ada to pay for the fee (the only source of ada is the minimum ada amount, which can't be reduced). The only other option is to burn the assets, which the requirements do not allow.

> Could dust ada end up here too?

Ada quantities that are less than or equal to `input_marginal_fee` will be omitted by the algorithm automatically, and not even considered for inclusion.

It is also possible for ada quantities greater than `input_marginal_fee` to be omitted: if the remaining UTxO set contains only **_n_** small values whose total value is less than (`minimum_ada_quantity` + `base_transaction_fee` + **_n_** × `input_marginal_fee`), then it won't be possible to create a transaction from them.